sklearn/feature_selection/_from_model.py

```diff
@@ -313,9 +326,17 @@ def fit(self, X, y=None, **fit_params):
                 )
 
         if self.prefit:
-            raise NotFittedError("Since 'prefit=True', call transform directly")
-        self.estimator_ = clone(self.estimator)
-        self.estimator_.fit(X, y, **fit_params)
+            try:
+                check_is_fitted(self.estimator)
+            except NotFittedError as exc:
+                raise NotFittedError(
+                    "When `prefit=True`, `estimator` is expected to be a fitted "
+                    "estimator."
+                ) from exc
+            self.estimator_ = deepcopy(self.estimator)
```

Comment:
We have not been consistent about this. I'm guessing you want to always deepcopy when an estimator when using `prefit`?

Reply:
Yep. We do that in stacking. In calibration, we fit a calibrated estimator so we don't need to make a deep copy. I think that this the behaviour that we want.

Reply:
It's not done in stacking yet:

https://github.com/scikit-learn/scikit-learn/blob/9258e329e05b75f4b531b6d0660c0b1bd4d20a16/sklearn/ensemble/_stacking.py#L184

I'm fine with changing stacking to deep copy to be consistent.

Reply:
Uhm you are right. The deep copy is on the CV object.

`SelectFromModel` will support `partial_fit`. It seems then important to take a deep copy to not modify the passed estimator.

In stacking, I don't think that anything would alter the `estimators`. So it might be unnecessary to enforce the deep copy there but it might be safer. We should work on SLEP017 :)

Reply:
Actually, I am remembering why I liked to have it in the first place. If we set the attribute of the external estimator, it will impact the internal estimator as well. The deep copy prevents this.